### ISSUE TITLE
Shell Out Of Window

### ISSUE THREAD

**Issue by vaibhawc** [pyzo/pyzo#559]
Hey I wanted to pull the shell out of the pyzo window and run it on attached monitor.
I have done such a thing before but cant do it on the latest version.
Could someone tell me if this functionality is still there and if yes how to do so.
Can't find it in the doc/readme/website/faq etc.

Thanks

**Comment by almarklein** [comment]
I am not sure whether this has ever worked, but it has not since at least a few years AFAIK :)

I remember that we've looked at this, but it was not trivial and prone to QT crashing. We could revisit it, but it's not high on my priority list to be honest...

**Comment by almarklein** [comment]
Dupe of #282 (but will close that one, since it has some unrelated info)

**Comment by vaibhawc** [comment]
If this can be done. It'd be really cool. Or if I could somehow attach a shell in the terminal to pyzo, which I think may be difficult. The purpose is to be able to able to use it on multiple monitors.

**Comment by vaibhawc** [comment]
Well, on this only, can I bind my linux terminal python shell to the pyzo editor. As if the not using pyzo's default shell interface to see outputs and run commands but that of my terminal. This will help me write an run command written in pyzo editor on that shell and I can use multiple shell windows to see which configuration of commands is best for me.

**Comment by almarklein** [comment]
Yes, that's a possibility, but it's not supported at the moment.

**Comment by vaibhawc** [comment]
Could you please hint on how it could be done so as I can make changes on my end?
I am ready to give time and energy for this.
Pyzo has been the only editor I ever used for python, if I can make this happen I might not have to switch to anything else.

**Comment by almarklein** [comment]
You'd have to look at `core/main.py`. This is where the widgets get initialized. All widgets get added to the dock in dockwidget-wrappers. I am not sure if a `QMainWindow`, with the right configuration is capable of putting its children in another window. If not, you'll want to put the shell in a new window (i.e. parent is `None` instead of putting it in the dock.

Out of curiosity, what is your objection of spreading the Pyzo window over both monitors?

**Comment by vaibhawc** [comment]
Not pyzo window, but I am trying to see multiple shell outputs in parallel and trying to compare different configurations to create code.
